python/cf_logging_enabled/cf_logging_enabled.py

```diff
@@ -0,0 +1,396 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  cf_logging_enabled
+Description:
+  Checks whether your CloudFront Distribution has been configured to store logs on an authorized S3 bucket
+Trigger:
+  Configuration change on AWS::CloudFront::Distribution
+Resource Type to report on:
+  AWS::CloudFront::Distribution
+Rule Parameters:
+  | ---------------------- | ---------- | -------------------------------------------------------------- |
+  | Parameter Name         | Type       | Description                                                    |
+  | ---------------------- | ---------- | -------------------------------------------------------------- |
+  | CentralLoggingBucket   | Mandatory  | Authorized S3 Bucket for storing CloudFront logs               |

```

Comment:
Why not making this parameter optional? I would allow to just verify that logging is enabled.

Reply:
I can make it optional. However, from the functionality perspective, if logs are not getting stored in proper bucket, then the monitoring / archival and other controls around logs might not get configured appropriately. As such, there would be no / very limited value of enabling the logging if the logs are not propoerly stored / monitored / archived. Hence, I think it should be a mandatory parameter. However, if it is a standard to make parameters optional, I am happy to do so.